### PR TITLE
Fixes batch priority

### PR DESCRIPTION
**Pull request by kenvac**
User Story: 12865

### CODE REVIEW

File: addons/udes_stock/wizard/stock_picking_to_batch.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        Args:
+            pickings (stock.picking): Stock picking object
+
+        Raises:
+            UserError: Raises exception when selected picking has different priority then batch
+        """
+        self.ensure_one()
+        picking_ids = self.env.context.get("active_ids")
+        pickings = self.env["stock.picking"].browse(picking_ids)
```
Comment (PeterC10) [inline addons/udes_stock/wizard/stock_picking_to_batch.py:20]: Nit: `self.env["stock.picking"]` should be in a variable at the start of the method, e.g.:
`Picking = self.env["stock.picking"]`

And then:
`pickings = Picking.browse(picking_ids)`

Comment (kenvac) [reply addons/udes_stock/wizard/stock_picking_to_batch.py]: done

File: addons/udes_stock/models/stock_picking_batch.py [hunk 2]
```diff
@@ -1268,6 +1268,30 @@ def reserve_pallet(self, pallet_name, picking=None):
...
+        user_name = self.env.user.name
+        u_log_batch_picking = warehouse.u_log_batch_picking
+
+        priority = self.priority
+        batch_name = self.name
+        diff_priority_pickings = pickings.filtered(lambda r: r.priority != priority).mapped("name")
+        if u_log_batch_picking:
+            for picking in pickings:
+                msg = _(
+                    "User: %s added picking %s with priority %s to batch %s with priority %s"
```
Comment (apatrick88) [inline addons/udes_stock/models/stock_picking_batch.py:1273]: How do we know whether this happened via the desktop or the mobile?

Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py:1273]: Do we want different message for mobile and desktop? 
If yes, then we need to pass context

**Review by PeterC10 — APPROVED** [review]

File: addons/udes_stock/models/stock_picking_batch.py (hunk 2 above)
Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py]: We are now passing variable to method

File: addons/udes_stock/models/stock_picking_batch.py [hunk 3]
```diff
@@ -1268,6 +1272,37 @@ def reserve_pallet(self, pallet_name, picking=None):
         else:
             self.write({"u_last_reserved_pallet_name": pallet_name})
 
+    def check_same_picking_priority(self, pickings, mode="mobile"):
+        """Checks if pickings priorities matches with batch priority
+
+        Args:
+            pickings (stock.picking): set of Picking objects
+        Return:
+            Boolean: Returns False if picking priority is different than batch else True
```
Comment (apatrick88) [inline addons/udes_stock/models/stock_picking_batch.py:1281]: Technically this function doesn't return a bool

Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py:1281]: Good pick. I changed the method but forgot to change the description. It returns the list of picking priority name

Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py]: done

File: addons/udes_stock/models/stock_picking_batch.py [hunk 4]
```diff
@@ -1278,14 +1281,14 @@ def check_same_picking_priority(self, pickings, mode="mobile"):
...
+            List: Returns list picking priority name which is different than batch priority
         """
         self.ensure_one()
-        warehouse = self.env.user.get_user_warehouse()
-        user_name = self.env.user.name
-        u_log_batch_picking = warehouse.u_log_batch_picking
+        u_log_batch_picking, user_name = self.get_log_batch_picking_flag()
 
-        priority = self.priority
+        # priority = self._origin.priority
```
Comment (apatrick88) [inline addons/udes_stock/models/stock_picking_batch.py:1289]: Nit: Commented out

File: addons/udes_stock/models/stock_picking_batch.py [hunk 5]
```diff
@@ -1303,6 +1306,16 @@ def check_same_picking_priority(self, pickings, mode="mobile"):
                 _logger.info(msg)
         return diff_priority_pickings
 
+    def get_log_batch_picking_flag(self):
+        """Get u_log_batch_picking configuration from warehouse and user name
+
+        Returns:
+            Boolean: u_log_batch_picking value
+        """
+        user = self.env["res.users"]
```
Comment (apatrick88) [inline addons/udes_stock/models/stock_picking_batch.py:1315]: Super nit: Capital User for an environment.

**Review by apatrick88 — APPROVED** [review]

File: addons/udes_stock/models/stock_picking_batch.py (hunk 4 above)
Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py]: done

File: addons/udes_stock/models/stock_picking_batch.py (hunk 5 above)
Comment (kenvac) [reply addons/udes_stock/models/stock_picking_batch.py]: done

**Pull request merged**